org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -3682,14 +3683,7 @@ else if (c == String.class)
 						sqlValues.append (encrypt(i,DB.TO_STRING ((String)value)));
 					else if (DisplayType.isLOB(dt))
 					{
-						if (p_info.isColumnMandatory(i))
-						{
-							sqlValues.append("''");		//	no db dependent stuff here -- at this point value is known to be not null
-						}
-						else
-						{
-							sqlValues.append("null");
-						}
+						sqlValues.append("decode('"+Hex.encodeHexString((byte[]) value)+"','hex')");
```

Comment:
is this just for Oracle?

Reply:
hi @CarlosRuiz-globalqss , decode(hex encoded data, 'hex') is actually a postgresql only call. need Oracle equivalent implementation here.

Reply:
i check that oracle has hextoraw to convert hex string to blob column but as i see the code. the sql will be same between pgsql and oracle. need a tweak to create different query for each database

Reply:
hi @uthadehikaru , you can do that by add a new method to the AdempiereDatabase interface and implement it in DB_PostgreSQL and DB_Oracle.

Reply:
i see. for temporary, this solution will only affect to postgresql script and for oracle, the value is null or empty.

Reply:
hi @hengsin , i check the Convert class and the database that actively convert is Convert_Postgresql. is idempiere use oracle sql first than when run on postgres, the convert class handle it? i think we can add convert method for blob query for each database.

Reply:
Don't do that, that's the legacy mode and we want to minimize the use of convert for postgresql ( the current default for postgresql is using that at the very minimal level). As I've posted, use the AdempiereDatabase interface instead and it is a very straightforward change.